### PR TITLE
fix(pypika translator): cannot select null in filterstep with operators in / nin / eq / ne [TCTC-7943]

### PR DESCRIPTION
**Pull request by fspot**
### in / nin

```sql
WHERE column IN (NULL, "foo", "bar")
```
does not work. Replace it with

```sql
WHERE column IS NULL OR column IN ("foo", "bar")
```

### eq / ne

```sql
WHERE column=NULL
```
does not work. Replace it with

```sql
WHERE column IS NULL
```

### CODE REVIEW

File: server/src/weaverbird/pipeline/steps/filter.py [hunk 1]
```diff
@@ -13,3 +13,8 @@ class FilterStep(BaseStep):
 
 class FilterStepWithVariables(FilterStep, StepWithVariablesMixin):
     condition: ConditionWithVariables  # type:ignore[assignment]
+
+    def model_dump(self, *, exclude_none: bool = False, **kwargs) -> dict[str, Any]:
+        # we don't want to drop fields with value=None like:
+        #     {'column': 'nullable_name', 'operator': 'eq', 'value': None}
+        return super().model_dump(exclude_none=exclude_none, **kwargs)
```
Comment (lukapeschke) [inline server/src/weaverbird/pipeline/steps/filter.py:20]: isn't `exclude_none=False` already the default behavior ? https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_dump

**Review by lukapeschke — APPROVED** [review]
Just a note on the new `model_dump` method, LGTM apart from that :+1:

File: server/src/weaverbird/pipeline/steps/filter.py (hunk 1 above)
Comment (fspot) [reply server/src/weaverbird/pipeline/steps/filter.py:20]: we have it overriden with exclude_none=True in BaseStep

Comment (lukapeschke) [reply server/src/weaverbird/pipeline/steps/filter.py:20]: :bulb:

**Pull request merged**